### PR TITLE
Updating minio-go library

### PR DESCRIPTION
**Pull request by vadmeste**
This PR includes latest minio-go fixes.

### CODE REVIEW

File: vendor/github.com/minio/minio-go/retry.go [hunk 1]
```diff
@@ -85,36 +85,32 @@ func (c Client) newRetryTimer(maxRetry int, unit time.Duration, cap time.Duratio
...
-				return true
-			} else if strings.Contains(err.Error(), "connection timed out") {
-				// If err is a net.Dial timeout, retry.
-				return true
-			} else if strings.Contains(err.Error(), "net/http: HTTP/1.x transport connection broken") {
-				// If error is transport connection broken, retry.
-				return true
-			}
+		}
+		if strings.Contains(err.Error(), "Connection closed by foreign host") {
```
Comment (ebozduman) [inline vendor/github.com/minio/minio-go/retry.go]: These `if` statements are `unreachable` since they are put inside the `switch` command, but not enclosed with a known `case` or `default` keyword, which are the only way to execute a code inside a `switch` command.
The problem will go away if the `if` statements are put inside a `default` block as it is in the old code.

File: vendor/github.com/minio/minio-go/utils.go [hunk 2]
```diff
@@ -223,6 +223,7 @@ var supportedHeaders = []string{
 	"content-disposition",
 	"content-language",
 	"x-amz-website-redirect-location",
+	"expires",
```
Comment (ebozduman) [inline vendor/github.com/minio/minio-go/utils.go]: How is this change related to this PR?

File: vendor/github.com/minio/minio-go/api-put-bucket.go [hunk 3]
```diff
@@ -178,6 +178,87 @@ func (c Client) removeBucketPolicy(bucketName string) error {
...
+		contentMD5Base64: sumMD5Base64(b),
+	}
+
+	// Execute PUT to upload a new bucket lifecycle.
+	resp, err := c.executeMethod(context.Background(), "PUT", reqMetadata)
+	defer closeResponse(resp)
+	if err != nil {
+		return err
+	}
+	if resp != nil {
```
[the remark this answers is not in the document]
Comment (harshavardhana) [reply vendor/github.com/minio/minio-go/api-put-bucket.go]: all of these should be addressed in minio-go @ebozduman open an issue there.
